### PR TITLE
Add Euler angle to RGB utility classes and AuxKernel

### LINKED ISSUE #6362: Add Euler angle to RGB color conversion utility

### PR DESCRIPTION
**Pull request by dschwen**
This adds some fixes to the code, namely:
- Use a `MooseEnum` for the `sd` parameter
- Data is provided by AuxVars and not Material properties
- Angle data is in degrees now!
- Mapping from 0..1 to 0..255 is fixed (and unit test re-golded)
- A test for `Euler2RGBAux` is added
- Minor code cleanup and documentation

Closes #6362

### CODE REVIEW

**Comment by dschwen** [comment]
This is an incompatible change with an internal application. A patch is ready.

**Comment by dschwen** [comment]
Ugh, header comments...

File: modules/phase_field/include/auxkernels/Euler2RGBAux.h [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+  virtual Real computeValue();
+
+private:
+  /// Reference direction of the sample
+  const unsigned int _sd;
+
+  ///@{ Euler angles to visualize
+  const VariableValue & _phi1;
+  const VariableValue & _phi;
+  const VariableValue & _phi2;
```
Comment (permcody) [inline modules/phase_field/include/auxkernels/Euler2RGBAux.h]: Are these the best names to use? (phi, phi1, phi2)

Comment (dschwen) [inline modules/phase_field/include/auxkernels/Euler2RGBAux.h]: Maybe not, but those are the standard names apparently.

Comment (permcody) [inline modules/phase_field/include/auxkernels/Euler2RGBAux.h]: There are still not enough emojis. Where's the "vomiting guy" emoji?

Comment (dschwen) [inline modules/phase_field/include/auxkernels/Euler2RGBAux.h]: I should say in the EBSD community. If we start changing this here we'll have to change it in the EBSD reader as well...

Comment (permcody) [inline modules/phase_field/include/auxkernels/Euler2RGBAux.h]: Yes I realize that, we should use standards where they exist. If if those standards are not the most well thought out.

Comment (dschwen) [inline modules/phase_field/include/auxkernels/Euler2RGBAux.h]: barface

**Comment by permcody** [comment]
:+1: Merged - waiting for Marmot patch

**Pull request merged**